src/anniversary/sf33rd/Source/Game/menu.c

```diff
@@ -2928,13 +2950,159 @@ u16 Check_Menu_Lever(u8 PL_id, s16 type) {
 
 INCLUDE_ASM("asm/anniversary/nonmatchings/sf33rd/Source/Game/menu", Suspend_Menu);
 
-INCLUDE_ASM("asm/anniversary/nonmatchings/sf33rd/Source/Game/menu", In_Game);
+void In_Game(struct _TASK *task_ptr) {
+    void (*In_Game_Jmp_Tbl[5])() = { Menu_Init, Menu_Select, Button_Config_in_Game, Character_Change, Pad_Come_Out };
+    In_Game_Jmp_Tbl[task_ptr->r_no[1]](task_ptr);
+}
+
+void Menu_Select(struct _TASK *task_ptr) {
+    s16 ix;
+    s16 oldy;
+
+    if (Check_Pad_in_Pause(task_ptr) == 0) {
+        switch (task_ptr->r_no[2]) {
+        case 0:
+            Pause_1st_Sub(task_ptr);
+            return;
+
+        case 1:
+            task_ptr->r_no[2]++;
+            Menu_Common_Init();
+            *Menu_Cursor_Y = Cursor_Y_Pos[0][0];
+            Menu_Suicide[0] = 0;
+            Menu_Suicide[1] = 0;
+            Menu_Suicide[2] = 0;
+            effect_10_init(0, 0, 0, 0, 0, 0x14, 0xC);
+            effect_10_init(0, 0, 2, 2, 0, 0x16, 0x10);
+            switch (Mode_Type) {
+            case 1:
+                effect_10_init(0, 0, 1, 5, 0, 0x10, 0xE);
+                return;
+
+            case 5:
+                effect_10_init(0, 0, 1, 4, 0, 0x15, 0xE);
+                return;
 
-INCLUDE_ASM("asm/anniversary/nonmatchings/sf33rd/Source/Game/menu", Menu_Select);
+            default:
+                effect_10_init(0, 0, 1, 1, 0, 0x11, 0xE);
+                return;
+            }
+            break;
+
+        case 2:
+            oldy = Menu_Cursor_Y[0];
+            IO_Result = MC_Move_Sub(Check_Menu_Lever(Pause_ID, 0), 0, 2, 0xFF);
+            switch (IO_Result) {
+
+            case 0x200:
+                task_ptr->r_no[2] = 0;
+                Menu_Suicide[0] = 1;
+                SE_selected();
+                return;
+
+            case 0x100:
+                switch (Menu_Cursor_Y[0]) {
+
+                case 0:
+                    task_ptr->r_no[2] = 0;
+                    Menu_Suicide[0] = 1;
+                    SE_selected();
+                    return;
+
+                case 1:
+                    SE_selected();
+                    switch (Mode_Type) {
+
+                    case 1:
+                        task_ptr->r_no[1] = 3U;
```

Comment:
Unsigned integer literals are usually redundant
```suggestion
                        task_ptr->r_no[1] = 3;
```